### PR TITLE
Add simple unit tests

### PR DESCRIPTION
**Pull request by nasdas-dev**
I have added some simple unit tests that will increase code coverage / branch coverage.
They contribute to issue #6207

- [ ] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)
- [x] Tests created for changes (if applicable)
- [ ] Manually tested changed features in running JabRef (always required)
- [ ] Screenshots added in PR description (for UI changes)
- [ ] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, submitted a pull request to the documentation repository.

### CODE REVIEW

**Comment by nasdas-dev** [comment]
will adjust and fix checkstyle this evening/tomorrow.

File: src/test/java/org/jabref/logic/journals/AbbreviationTest.java [hunk 1]
```diff
@@ -101,4 +103,17 @@ void testDefaultAndShortestUniqueAbbreviationsAreSame() {
...
+    @Test
+    void testToString() {
+      Abbreviation abbreviation = new Abbreviation("Long Name", "L N", "LN");
+      assertEquals(abbreviation.toString(), "Abbreviation{name=Long Name, abbreviation=L N, medlineAbbreviation=L N, shortestUniqueAbbreviation=LN}");
+    }
+
+    @Test
+    void testEquals() {
+      Abbreviation abbreviation = new Abbreviation("Long Name", "L N", "LN");
+      assertTrue(abbreviation.equals(abbreviation));
```
Comment (ellieMayVelasquez) [inline src/test/java/org/jabref/logic/journals/AbbreviationTest.java:116]: I detect that this code is problematic. According to the [Correctness (CORRECTNESS)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#correctness-correctness), [SA: Self comparison of value with itself (SA_LOCAL_SELF_COMPARISON)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#sa-self-comparison-of-value-with-itself-sa-local-self-comparison).
This method compares a local variable with itself, and may indicate a typo or a logic error. Make sure that you are comparing the right things.

Comment (nasdas-dev) [reply src/test/java/org/jabref/logic/journals/AbbreviationTest.java]: Thanks for the feedback! This test originally was meant to cover the branch in the implementation of the equals method, where it returns true for being compared to itself. But seeing your linked references, I agree that it makes more sense to leave it out - I have adjusted the test now.

Comment (ellieMayVelasquez) [reply src/test/java/org/jabref/logic/journals/AbbreviationTest.java]: Happy to help :)

File: src/test/java/org/jabref/gui/autocompleter/SuggestionProvidersTest.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+      entry.setField(personEntryField, "Goethe");
+      entry.setField(singleEntryField, "Single");
+      entry.setField(multipleEntryField, "Multiple");
+      entry.setField(journalEntryField, "Journal");
+      entry.setField(publisherEntryField, "Publisher");
+      entry.setField(specialEntryField, "2000");
+      database.insertEntry(entry);
+
+      assertSame("org.jabref.gui.autocompleter.EmptySuggestionProvider", empty.getForField(personEntryField).getClass().getName());
```
[the remark this answers is not in the document]
Comment (koppor) [reply src/test/java/org/jabref/gui/autocompleter/SuggestionProvidersTest.java:41]: (applies for all lines here)

File: src/test/java/org/jabref/logic/citationkeypattern/CitationKeyGeneratorTest.java [hunk 3]
```diff
@@ -747,6 +747,9 @@ void testLastPage() {
         assertEquals("97", CitationKeyGenerator.lastPage("7,41,73--97"));
         assertEquals("97", CitationKeyGenerator.lastPage("7,41,97--73"));
         assertEquals("43", CitationKeyGenerator.lastPage("43+"));
+        assertEquals("0", CitationKeyGenerator.lastPage("00--0"));
+        assertEquals("1", CitationKeyGenerator.lastPage("1--1"));
+
```
Comment (koppor) [inline src/test/java/org/jabref/logic/citationkeypattern/CitationKeyGeneratorTest.java:752]: Please no empty line at the end of the method

Comment (koppor) [reply src/test/java/org/jabref/logic/citationkeypattern/CitationKeyGeneratorTest.java:752]: Please convert the whole test to a `@ParameterizedTest`. See https://github.com/JabRef/jabref/pull/7544 for an inspiration

File: src/test/java/org/jabref/logic/citationstyle/CitationStyleCacheTest.java [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+  private BibDatabaseContext databaseContext;
+  private CitationStyleCache csCache;
+
+  @Test
+  void getCitationForTest() {
+    BibEntry bibEntry = new BibEntry();
+    bibEntry.setCitationKey("test");
+    List<BibEntry> entries = new ArrayList<>();
+    entries.add(0, bibEntry);
+    BibDatabase database = new BibDatabase(entries);
```
Comment (koppor) [inline src/test/java/org/jabref/logic/citationstyle/CitationStyleCacheTest.java:28]: Can't one do the following?

```suggestion
    BibDatabase database = new BibDatabase(List.of(new BibEntry("test"));
```

File: src/test/java/org/jabref/logic/citationstyle/CitationStyleCacheTest.java [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+    BibDatabaseContext databaseContext = new BibDatabaseContext(database);
+    CitationStyleCache csCache = new CitationStyleCache(databaseContext);
+
+    assertNotNull(csCache.getCitationFor(bibEntry));
```
Comment (koppor) [inline src/test/java/org/jabref/logic/citationstyle/CitationStyleCacheTest.java:32]: Please test for a concrete result.

File: src/test/java/org/jabref/logic/formatter/casechanger/UpperCaseFormatterTest.java [hunk 6]
```diff
@@ -24,6 +24,9 @@ public void test() {
         assertEquals("UPPER", formatter.format("UPPER"));
         assertEquals("UPPER {lower}", formatter.format("upper {lower}"));
         assertEquals("UPPER {l}OWER", formatter.format("upper {l}ower"));
+        assertEquals("1", formatter.format("1"));
```
Comment (koppor) [inline src/test/java/org/jabref/logic/formatter/casechanger/UpperCaseFormatterTest.java:27]: Please convert to `@ParameterizedTests`

File: src/test/java/org/jabref/logic/journals/AbbreviationTest.java [hunk 7]
```diff
@@ -101,4 +103,18 @@ void testDefaultAndShortestUniqueAbbreviationsAreSame() {
...
+    void testToString() {
+      Abbreviation abbreviation = new Abbreviation("Long Name", "L N", "LN");
+      assertEquals(abbreviation.toString(), "Abbreviation{name=Long Name, abbreviation=L N, medlineAbbreviation=L N, shortestUniqueAbbreviation=LN}");
+    }
+
+    @Test
+    void testEquals() {
+      Abbreviation abbreviation = new Abbreviation("Long Name", "L N", "LN");
+      Abbreviation otherAbbreviation = new Abbreviation("Long Name", "L N", "LN");
+      assertTrue(abbreviation.equals(otherAbbreviation));
```
Comment (koppor) [inline src/test/java/org/jabref/logic/journals/AbbreviationTest.java:117]:
```suggestion
      assertEquals(abbreviation, otherAbbreviation);
```

File: src/test/java/org/jabref/logic/journals/AbbreviationTest.java [hunk 8]
```diff
@@ -101,4 +103,18 @@ void testDefaultAndShortestUniqueAbbreviationsAreSame() {
...
+      assertFalse(abbreviation.equals("String"));
```
Comment (koppor) [inline src/test/java/org/jabref/logic/journals/AbbreviationTest.java:118]:
```suggestion
      assertEquals("String", abbreviation);
```

File: src/test/java/org/jabref/model/FieldChangeTest.java [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+
+class FieldChangeTest {
+
+  private FieldChange Fieldchange;
+
+  @Test
+  void testHashCode() {
+    FieldChange fcNull = new FieldChange(null, null, null, null);
+    assertEquals(923521, fcNull.hashCode());
+  }
```
Comment (koppor) [inline src/test/java/org/jabref/model/FieldChangeTest.java:21]: Please no test for `hashCode`. 

You can include https://jqno.nl/equalsverifier/ as dependency and work on that. Please do not test for that manually.

File: src/test/java/org/jabref/model/FieldChangeTest.java [hunk 10]
```diff
@@ -0,0 +1,63 @@
...
+
+  @Test
+  void testEquals() {
+    BibEntry entry = new BibEntry();
```
Comment (koppor) [inline src/test/java/org/jabref/model/FieldChangeTest.java:25]: This is a very hard to read test. Please use https://jqno.nl/equalsverifier/ or just delete the test.

I never used https://jqno.nl/equalsverifier/ for myself. Please add a proposal based on that and we can investigate whether this is really worth it or causes much trouble.

File: src/test/java/org/jabref/model/FieldChangeTest.java [hunk 11]
```diff
@@ -0,0 +1,63 @@
...
+    assertFalse(fcBlankAll.equals(fc));
+    assertFalse(fc.equals(fcOther));
+    assertFalse(fcBlankField.equals(fc));
+    assertFalse(fcBlankOldValue.equals(fc));
+    assertFalse(fcBlankNewValue.equals(fc));
+    assertTrue(fcBlankAll.equals(fcBlankAll));
+  }
+
+  @Test
+  void testToString() {
```
Comment (koppor) [inline src/test/java/org/jabref/model/FieldChangeTest.java:53]: Even though "Effective Java" recomments that `toString` is a contract; JabRef does not rely on that in this case. Please remove that test.

File: src/test/java/org/jabref/logic/journals/AbbreviationTest.java [hunk 12]
```diff
@@ -101,4 +103,18 @@ void testDefaultAndShortestUniqueAbbreviationsAreSame() {
         Abbreviation abbreviation = new Abbreviation("Long Name", "L N");
         assertEquals(abbreviation.getAbbreviation(), abbreviation.getShortestUniqueAbbreviation());
     }
+
+    @Test
+    void testToString() {
```
Comment (koppor) [inline src/test/java/org/jabref/logic/journals/AbbreviationTest.java:108]: JabRef does not rely on the `toString` contract in this case. Please remove the test.

If you want, you can check for nonNull of `toString`.

File: src/test/java/org/jabref/model/entry/EntryLinkListTest.java [hunk 13]
```diff
@@ -29,6 +31,7 @@ public void before() {
         link = links.get(0);
         source = create("source");
         target = create("target");
+        entry = create("entry");
```
Comment (koppor) [inline src/test/java/org/jabref/model/entry/EntryLinkListTest.java:34]: Please do not initalize the variable here. It is used only in **one** test. Just move the variable to `givenBibEntryWhenParsingThenExpectLink`.

**Review by koppor — CHANGES_REQUESTED** [review]
Some comments in there. The main comment is that `toString` doesn't need to be tested if the software does not rely on that contract.

**Comment by koppor** [comment]
I think your course is over @ellieMayVelasquez, thus no more activity here?

**Comment by nasdas-dev** [comment]
> I think your course is over @ellieMayVelasquez, thus no more activity here?

Thanks a lot for all the feedback @koppor ! - I will resolve all of these issues within this month hopefully, just haven't gotten around to it yet due to lots of parallel coursework atm :)

**Comment by koppor** [comment]
@nasdas-dev May I ask which month you meant? 😇

**Review by Siedlerchr — APPROVED** [review]

**Pull request merged**